lib/src/dart2_suggestors/props_and_state_mixin_meta_remover.dart

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:analyzer/analyzer.dart';
+import 'package:codemod/codemod.dart';
+
+import '../constants.dart';
+
+/// Suggestor that removes the static `meta` field from every props and state
+/// mixin class if it was added by the backwards-compatible version of the
+/// Dart 2 upgrade codemod.
+///
+/// The meta field was only necessary for Dart 1 compatibility; if props and
+/// state mixin classes are prefixed with `_$`, the builder will generate the
+/// prefix-stripped version that will contain this static meta field.
+class PropsAndStateMixinMetaRemover extends RecursiveAstVisitor
+    with AstVisitingSuggestorMixin
+    implements Suggestor {
+  @override
+  visitClassDeclaration(ClassDeclaration node) {
+    // Only looking for @PropsMixin() and @StateMixin() classes.
+    final isOverReactMixin = node.metadata.any((annotation) {
+      return overReactMixinAnnotationNames.contains(annotation.name.name);
+    });
+    if (!isOverReactMixin) {
+      return;
+    }
+
+    final metaField = node.getField('meta');
+    if (metaField == null || metaField.initializer == null) {
```

Comment:
Yeah the only problem is that I don't have a good way to warn right now since the interactive codemod runner clears the terminal for each suggestion. I think it's worth looking into supporting that in dart_codemod, but I'm not sure if it's worth it right now. For our own use, we'll be going through the transitional stage which should illuminate conflicts anyway